frontend/src/pages/GamePage.tsx

```diff
@@ -1,99 +1,200 @@
-            <br />,
-            <TerminalOutput>
-                {`Updated stats - Health: ${stats.health + selectedOption.impact.health}, Happiness: ${
-                    stats.happiness + selectedOption.impact.happiness
-                }, Money: ${stats.money + selectedOption.impact.money}.`}
-            </TerminalOutput>,
-            <br />,
-        ]);
-
-        // Move to the next question
-        const nextQuestionIndex = currentQuestionIndex + 1;
-        if (nextQuestionIndex < quizQuestions.length) {
-            setCurrentQuestionIndex(nextQuestionIndex);
-            const nextQuestion = quizQuestions[nextQuestionIndex];
-            setTerminalLineData((prevData) => [
-                ...prevData,
-                <br />,
-                <TerminalOutput>{`Question ${nextQuestionIndex + 1}: ${nextQuestion.question}`}</TerminalOutput>,
-                ...nextQuestion.options.map((option, index) => (
-                    <TerminalOutput key={`t-${index}`}>{`${index + 1}. ${option.text}`}</TerminalOutput>
-                )),
-            ]);
+            firstQuestionText,
+            firstOptionsText,
+          ]);
         } else {
-            setTerminalLineData((prevData) => [
-                ...prevData,
-                <TerminalOutput>
-                    {`Game Over! Final stats - Health: ${stats.health}, Happiness: ${stats.happiness}, Money: ${stats.money}.`}
-                </TerminalOutput>,
-            ]);
+          setGameOver(true);
+          setTerminalLineData((prevData) => [...prevData, "No questions available. Game over!"]);
         }
+      } catch (err) {
+        setError(err instanceof Error ? err.message : "An error occurred.");
+      } finally {
+        setIsLoading(false);
+      }
     };
 
-    return (
-        <div className="container">
-            <Terminal
-                name="Life Choices Game"
-                colorMode={ColorMode.Dark}
-                onInput={handleInput}
-            >
-                {terminalLineData}
-            </Terminal>
-        </div>
+    fetchQuestions();
+  }, []);
+
+  const submitAnswer = async (
+    questionId: string,
+    selectedOption: string,
+    newStats: { health: number; wealth: number; happiness: number }
+  ): Promise<boolean> => {
+    try {
+      const response = await fetch("https://kernel-panic.onrender.com/api/submitAnswer", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          userId,
+          answeredQuestion: {
+            questionId,
+            selectedOption,
+          },
+          updatedStats: newStats, // Sending updated stats to backend
+        }),
+      });
+
+      const result = await response.json();
+
+      if (response.ok) {
+        return true; // Successfully submitted answer
+      } else {
+        throw new Error(result.message || "Failed to submit answer.");
+      }
```

Comment:
These all things can be simplified if you use PanicApi, checkout `LandingPage.tsx` for reference, same goes with every api call you make in the project